### PR TITLE
Add merge by squash option

### PR DESCRIPTION
**Pull request by Soullivaneuh**
Refs:

- https://developer.github.com/v3/pulls/#merge-a-pull-request-merge-button
- https://developer.github.com/changes/2016-04-01-squash-api-preview/

### CODE REVIEW

File: lib/Github/Api/PullRequest.php [hunk 1]
```diff
@@ -115,10 +115,18 @@ public function merged($username, $repository, $id)
...
-    public function merge($username, $repository, $id, $message, $sha)
+    public function merge($username, $repository, $id, $message, $sha, $squash = false, $title = null)
     {
-        return $this->put('repos/'.rawurlencode($username).'/'.rawurlencode($repository).'/pulls/'.rawurlencode($id).'/merge', array(
-            'commit_message' => $message, 'sha' => $sha
-        ));
+        $params = array(
+            'commit_message' => $message,
+            'sha' => $sha,
+            'squash' => $squash,
```
Comment (GrahamCampbell) [inline lib/Github/Api/PullRequest.php]: Does this break if not using the preview API?

Comment (GrahamCampbell) [inline lib/Github/Api/PullRequest.php]: Or does GitHub ignore the invalid extra field?

Comment (Soullivaneuh) [inline lib/Github/Api/PullRequest.php]: If you don't use the preview API, GitHub will just make a classic merge. :+1:

Comment (GrahamCampbell) [inline lib/Github/Api/PullRequest.php]: It would be safer to only pass through `squash` if it's set to true, otherwise omit it, wouldn't it?

Comment (Soullivaneuh) [inline lib/Github/Api/PullRequest.php]: This would change nothing. If the parameter is not managed by GitHub API v3, it is simply ignored.

**Comment by Soullivaneuh** [comment]
@cursedcoder I fixed the tests. :+1:

**Comment by Soullivaneuh** [comment]
Thanks! Any ETA for the 1.7? :smile:

**Comment by GrahamCampbell** [comment]
Does this actually work for you? I can't get it to work for me. Yes, I am using the `application/vnd.github.polaris-preview+json` accept. :)

**Comment by Soullivaneuh** [comment]
@GrahamCampbell Yes.

See my GitHub client init: https://github.com/sonata-project/dev-kit/blob/87f5a3cf59ab9f9b5dabdb95f5eac5b6574641f4/src/Console/Command/AbstractCommand.php#L88-L95

And the merge call: https://github.com/sonata-project/dev-kit/blob/87f5a3cf59ab9f9b5dabdb95f5eac5b6574641f4/src/Console/Command/PullRequestAutoMergeCommand.php#L121-L129

**Comment by GrahamCampbell** [comment]
nvm

**Pull request merged**